File: AppStore_InfoApp.class.js
class AppStoreInfoApp {
  async sentRequest(bundleid) {
    const response = await fetch(`http://itunes.apple.com/lookup?bundleId=${bundleid}`);
    const data = await response.json();
    return data.results[0];
  }

  async getInfoApp(bundleid, Version) {
    const data = await this.sentRequest(bundleid);
    const V_AppStore = data.version;
    const appLink = data.trackViewUrl;

    if (V_AppStore === Version) {
      return this.responseApi('Yes', appLink);
    } else {
      return this.responseApi('No', appLink);
    }
  }

  responseApi(Status, url) {
    if (Status === 'Yes') {
      const json = {
        status: 'success',
        status_message: 'There is no application update',
      };
      return JSON.stringify(json);
    } else {
      const json = {
        status: 'failed',
        status_message: 'The version does not match. An update is required',
        app_link: url,
      };
      return JSON.stringify(json);
    }
  }
}

// Example usage:
const app = new AppStoreInfoApp();
app.getInfoApp('your_bundle_id', 'your_version')
  .then(response => console.log(response))
  .catch(error => console.error(error));

File: AppStore_InfoApp.class.py
// Developer By : Azozz ALFiras

import json
import requests

class AppStoreInfoApp:
    def sent_request(self, bundleid):
        url = f"http://itunes.apple.com/lookup?bundleId={bundleid}"
        response = requests.get(url)
        data = response.json()
        return data["results"][0]

    def get_info_app(self, bundleid, version):
        data = self.sent_request(bundleid)
        V_AppStore = data["version"]
        app_link = data["trackViewUrl"]

        if V_AppStore == version:
            return self.response_api('Yes', app_link)
        else:
            return self.response_api('No', app_link)

    def response_api(self, status, url):
        if status == 'Yes':
            json_data = {
                'status': 'success',
                'status_message': 'There is no application update'
            }
        else:
            json_data = {
                'status': 'failed',
                'status_message': 'The version does not match. An update is required',
                'app_link': url
            }

        return json.dumps(json_data)

# Example usage:
app = AppStoreInfoApp()
result = app.get_info_app('your_bundle_id', 'your_version')
print(result)
